src/cavity.jl

```diff
@@ -0,0 +1,108 @@
+@kwdef mutable struct RFParams{T<:Number} <: AbstractParams
+    frequency::T          = Float32(0.0) # RF frequency in Hz or Harmonic number
+    voltage::T            = Float32(0.0) # Voltage in V 
+    phi0::T               = Float32(0.0) # Phase at reference energy
+    const harmon_master::Bool = false    # false = frequency in Hz, true = harmonic number
+
+    function RFParams(args...)
+        return new{promote_type(map(x->typeof(x),args)...)}(args...)
+    end
+end
+
+Base.eltype(::RFParams{T}) where {T} = T
+Base.eltype(::Type{RFParams{T}}) where {T} = T
+
+
+function Base.isapprox(a::RFParams, b::RFParams) 
+    return  a.frequency     ≈  b.frequency && 
+            a.voltage       ≈  b.voltage && 
+            a.phi0          ≈  b.phi0 && 
+            a.harmon_master == b.harmon_master
```

Comment:
```suggestion
    return 
            a.harmon_master == b.harmon_master &&
            a.frequency     ≈  b.frequency && 
            a.voltage       ≈  b.voltage && 
            a.phi0          ≈  b.phi0         
```
Immediately short circuit conditional if `harmon_master`s disagree